### PR TITLE
add system command to get and set system time

### PR DESCRIPTION
**Pull request by baumanta**
This PR adds a nuttx shell command to get and set the system time

image

### CODE REVIEW

**Comment by baumanta** [comment]
@bkueng currently this is only added for the v5x target. Do we want to add it for others as well? If so for which ones?

**Comment by dagar** [comment]
> @bkueng currently this is only added for the v5x target. Do we want to add it for others as well? If so for which ones?

I would add it to all of them where it works (most NuttX boards) and leave it commented out in the flash constrained boards like fmu-v2. Skipping it now is also fine and I'll propagate it everywhere the next time I do a pass to sync all the configurations.

**Comment by dagar** [comment]
> > @bkueng currently this is only added for the v5x target. Do we want to add it for others as well? If so for which ones?
> 
> I would add it to all of them where it works (most NuttX boards)

Nevermind, looks like it should be fine to include everywhere (linux, etc).

File: boards/holybro/kakutef7/default.cmake [hunk 1]
```diff
@@ -68,6 +68,7 @@ px4_add_board(
 		reboot
 		reflect
 		sd_bench
+		system_time
```
Comment (bkueng) [inline boards/holybro/kakutef7/default.cmake:71]: I'd keep it out here as we're also limited on FLASH.

File: src/systemcmds/system_time/system_time.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	if (argc >= 2) {
+		if (!strcmp(argv[1], "get")) {
+			//get system time
+			struct timespec ts = {};
+			px4_clock_gettime(CLOCK_REALTIME, &ts);
+			time_t utc_time_sec = ts.tv_sec + (ts.tv_nsec / 1e9);
+
+			//convert to date time
+			char buf[80];
+			strftime(buf, sizeof(buf), "%a %Y-%m-%d %H:%M:%S %Z", localtime(&utc_time_sec));
```
Comment (bkueng) [inline src/systemcmds/system_time/system_time.cpp:57]: Can you use `localtime_r` which is thread-safe?

File: src/systemcmds/system_time/system_time.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+			//get time since boot
+			hrt_abstime since_boot_sec = hrt_absolute_time() / 1_s;
+
+			PX4_INFO("Unix epoch time: %d", utc_time_sec);
```
Comment (bkueng) [inline src/systemcmds/system_time/system_time.cpp:62]:
```suggestion
			PX4_INFO("Unix epoch time: %ld", (long)utc_time_sec);
```

**Review by bkueng — COMMENTED** [review]
Thanks for going through all the boards!

File: src/systemcmds/system_time/system_time.cpp [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+			PX4_INFO("System time: %s", buf);
+			PX4_INFO("Uptime (since boot): %" PRIu64 " s", since_boot_sec);
+			return 0;
+		}
+
+		if (!strcmp(argv[1], "set")) {
+			if (argc == 3) {
+				//set system time
+				struct timespec ts = {};
+				ts.tv_sec = (time_t)strtoul(argv[2], NULL, 0);
```
Comment (dagar) [inline src/systemcmds/system_time/system_time.cpp:74]:
```suggestion
				ts.tv_sec = (time_t)strtoul(argv[2], nullptr, 0);
```

Screenshot from 2020-09-25 10-33-07

**Review by dagar — APPROVED** [review]

**Pull request merged**